### PR TITLE
[SDP-923] events/cmd/data: add patch anchor transaction completion event handler

### PR DESCRIPTION
**Pull request by CaioTeixeira95**
### What

This PR adds the `PatchAnchorPlatformTransactionCompletionEventHandler` to handle the events produced by TSS to update transactions on the Anchor Platform.

### Why

We need to update the transactions/payments on the Anchor Platform as soon as they are processed by TSS.

### Known limitations

N/A

#### PR Structure

* [x] This PR has a reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR title and description are clear enough for anyone to review it.
* [x] This PR does not mix refactoring changes with feature changes (split into two PRs otherwise).

#### Thoroughness

* [x] This PR adds tests for the new functionality or fixes.
* [x] This PR contains the link to the Jira ticket it addresses.

#### Configs and Secrets

* [x] No new CONFIG variables are required -OR- the new required ones were added to the helmchart's [`values.yaml`] file.
* [x] No new CONFIG variables are required -OR- the new required ones were added to the deployments ([`pr-preview`], [`dev`], [`demo`], `prd`).
* [x] No new SECRETS variables are required -OR- the new required ones were mentioned in the helmchart's [`values.yaml`] file.
* [x] No new SECRETS variables are required -OR- the new required ones were added to the deployments ([`pr-preview secrets`], [`dev secrets`], [`demo secrets`], `prd secrets`).

#### Release

* [x] This is not a breaking change.
* [ ] **This is ready for production.**. If your PR is not ready for production, please consider opening additional complementary PRs using this one as the base. Only merge this into `develop` or `main` after it's ready for production!

#### Deployment

* [ ] Does the deployment work after merging?

[`values.yaml`]: ../helmchart/sdp/values.yaml
[`pr-preview`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/stellar-disbursement-platform/backend-helm-values
[`dev`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/backend-helm-values
[`demo`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-backend-helm-values
[`pr-preview secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/externalsecrets-common-previews.yaml#L241-L346
[`dev secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/stellar-disbursement-platform-externalsecrets.yaml
[`demo secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-sdp-externalsecrets.yaml

### CODE REVIEW

File: internal/events/eventhandlers/patch_anchor_platform_transaction_completion_event_handler.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+		crashTrackerClient: options.CrashTrackerClient,
+	}
+}
+
+func (h *PatchAnchorPlatformTransactionCompletionEventHandler) Name() string {
+	return "PatchAnchorPlatformTransactionCompletionEventHandler"
+}
+
+func (h *PatchAnchorPlatformTransactionCompletionEventHandler) CanHandleMessage(ctx context.Context, message *events.Message) bool {
+	return message.Topic == events.PatchAnchorPlatformTransactionCompletionTopic
```
Comment (marwen-abid) [inline internal/events/eventhandlers/patch_anchor_platform_transaction_completion_event_handler.go:51]: I think it's possible to use one topic for this handler and the one that synchronizes TSS txs back to the SDP. 
Meaning having a `payments.payment_completed` topic that both event handlers can read from

**Review by marwen-abid — APPROVED** [review]
LGTM! This is looking great Caio 🚀

**Comment by CaioTeixeira95** [comment]
@marwen-abid @marcelosalloum I've marked this PR as a draft again to fix the tests and add the new logic using the same topic for both `PaymentFromSubmitterEventHandler` and this one. Thank you for the review and requested changes till now! 🚀

**Comment by CaioTeixeira95** [comment]
@marcelosalloum @marwen-abid ready for review again! 💪

**Review by marwen-abid — APPROVED** [review]
This looks amazing @CaioTeixeira95 🚀 🚀 Thank you for addressing the comments on this.

**Pull request merged**